src/chunked_scatter/chunked_scatter.py

```diff
@@ -0,0 +1,165 @@
+# SOFTWARE.
+
+import argparse
+from pathlib import Path
+from typing import TextIO
+
+
+def dict_chunker(in_file: TextIO, chunk_size: int, overlap: int):
+    """
+    Chunk the chromosomes/contigs from a dict.
+    :param in_file: The input file.
+    :param chunk_size: The size of the chunks.
+    :param overlap: The size of the overlap between chunks.
+    """
+    for line in in_file.readlines():
+        line = line.split()
+        if line[0] == "@SQ":
+            for field in line:
+                if field[:2] == "LN":
+                    length = int(field.split(":")[1])
+                elif field[:2] == "SN":
+                    name = ":".join(field.split(":")[1:])
+            # This will cause the last chunk to be between 0.5 and 1.5
+            # times the chunk_size in length, this way we avoid the
+            # possibility that the last chunk ends up being to small
+            # (eg. 1+overlap bases).
+            position = 0
+            while position + chunk_size*1.5 < length:
+                if position-overlap <= 0:
+                    yield [name, 0, int(position+chunk_size)]
+                else:
+                    yield [name, int(position-overlap), int(position+chunk_size)]
+                position += chunk_size
+            if position-overlap <= 0:
+                yield [name, 0, length]
+            else:
+                yield [name, int(position-overlap), length]
+
+
+def bed_chunker(in_file: TextIO, chunk_size: int, overlap: int):
+    """
+    Chunk the entries of the bed file.
+    :param in_file: The input file.
+    :param chunk_size: The size of the chunks.
+    :param overlap: The size of the overlap between chunks.
+    """
+    for line in in_file.readlines():
+        line = line.strip().split("\t")
+        if line[0] not in ["browser", "track"] and len(line) >= 3:
+            start = int(line[1])
+            end = int(line[2])
+            name = line[0]
+            position = start
+            # This will cause the last chunk to be between 0.5 and 1.5
+            # times the chunk_size in length, this way we avoid the
+            # possibility that the last chunk ends up being to small
+            # (eg. 1+overlap bases).
+            while position + chunk_size*1.5 < end:
+                if position-overlap <= start:
+                    yield [name, start, int(position+chunk_size)]
+                else:
+                    yield [name, int(position-overlap),
+                           int(position+chunk_size)]
+                position += chunk_size
+            if position-overlap <= start:
+                yield [name, start, end]
+            else:
+                yield [name, int(position-overlap), end]
+
+
+def main():
+    args, bed_input = parse_args()
+    current_scatter = 0
+    current_scatter_size = 0
+    current_contig = None
+    out_file = open("{}{}.bed".format(args.prefix, current_scatter), "w")
```

Comment:
When your application crashes there will still be an open file handle. Best to use a context manager with open for this as well.
Other option: use try: finally: . Anything in finally will always be executed, even if there is an exception in the try block.

Reply:
Python's cleanup at shutdown should always close open file handles. Given that out_file changes, I'm not sure that a context manager would work here, a try statement might...